code/modules/power/singularity/narsie.dm

```diff
@@ -66,34 +66,68 @@
 		if(player.stat != DEAD && player.loc && is_station_level(player.loc.z) && !iscultist(player) && !isanimal(player))
 			souls_needed[player] = TRUE
 	soul_goal = round(1 + LAZYLEN(souls_needed) * 0.75)
-	INVOKE_ASYNC(src, .proc/begin_the_end)
+	INVOKE_ASYNC(GLOBAL_PROC, .proc/begin_the_end)
 
-/obj/singularity/narsie/large/cult/proc/begin_the_end()
+/proc/begin_the_end()
 	sleep(50)
+	if(QDELETED(GLOB.cult_narsie)) // uno
+		priority_announce("Status report? We detected a anomaly, but it disappeared almost immediately.","Central Command Higher Dimensional Affairs", 'sound/misc/notice1.ogg')
+		GLOB.cult_narsie.resolved = TRUE
+		sleep(20)
+		INVOKE_ASYNC(GLOBAL_PROC, .proc/cult_ending_helper, 2)
+		return
 	priority_announce("An acausal dimensional event has been detected in your sector. Event has been flagged EXTINCTION-CLASS. Directing all available assets toward simulating solutions. SOLUTION ETA: 60 SECONDS.","Central Command Higher Dimensional Affairs", 'sound/misc/airraid.ogg')
 	sleep(500)
+	if(QDELETED(GLOB.cult_narsie)) // dos
+		priority_announce("Simulations aborted, sensors report that the acasual event is normalizing. Good work, crew.","Central Command Higher Dimensional Affairs", 'sound/misc/notice1.ogg')
+		GLOB.cult_narsie.resolved = TRUE
+		sleep(20)
+		INVOKE_ASYNC(GLOBAL_PROC, .proc/cult_ending_helper, 2)
+		return
 	priority_announce("Simulations on acausal dimensional event complete. Deploying solution package now. Deployment ETA: ONE MINUTE. ","Central Command Higher Dimensional Affairs")
 	sleep(50)
 	set_security_level("delta")
-	SSshuttle.registerHostileEnvironment(src)
+	SSshuttle.registerHostileEnvironment(GLOB.cult_narsie)
 	SSshuttle.lockdown = TRUE
 	sleep(600)
-	if(resolved == FALSE)
-		resolved = TRUE
+	if(QDELETED(GLOB.cult_narsie)) // tres
+		priority_announce("Normalization detected! Abort the solution package!","Central Command Higher Dimensional Affairs", 'sound/misc/notice1.ogg')
+		GLOB.cult_narsie.resolved = TRUE
+		sleep(20)
+		set_security_level("red")
+		SSshuttle.clearHostileEnvironment(GLOB.cult_narsie)
```

Comment:
`QDELETED(GLOB.cult_narsie)` is true in this branch, meaning `GLOB.cult_narsie` is likely to be null

Intentionally letting it not be null is a bad idea, you shouldn't hang on to references to something after its Destroy

Reply:
This is fixed now.